components/net_traits/bluetooth_thread.rs

```diff
@@ -85,20 +85,19 @@ pub enum BluetoothRequest {
 
 #[derive(Deserialize, Serialize)]
 pub enum BluetoothResponse {
-    RequestDevice(BluetoothDeviceMsg),
-    GATTServerConnect(bool),
+    RequestDevice(BluetoothResult<BluetoothDeviceMsg>),
```

Comment:
Given that every response includes a result value, it probably makes more sense to use `BluetoothResult<BluetoothResponse>` instead. This will mean less code duplication for handling errors in places like bluetoothremotegattcharacteristic.rs. It could even be handled inside BluetoothResponseListener::response, so none of the rest of the implementations of AsyncBluetoothListener need to care about errors.